# Linear models

## Naoya's book

Naoya read a book cover to cover in a single session, at a rate of $55$ pages per hour. After reading for $4$ hours, he had $330$ pages left to read.

**How many pages is the book?**

**How many hours did it take Naoya to read the entire book?**

Naoya read $55$ pages per hour, so he read $55T$ pages in $T$ hours.

The total number of pages is comprised of the number of pages Naoya had already read and the number of pages that remained. We can express this with the equation $B=55T+R$, where:

* $B$ represents the book's length
* $T$ represents the time (in hours)
* $R$ represents the number of remaining pages to read at a given time

In [2]:
from sympy import *
B, T, R = symbols('B T R')
eq = Eq(B, 55 * T + R)

eq.subs([(T, 4), (R, 330)])

Eq(B, 550)

We know that after reading for $4$ hours $(T={4})$, Naoya had $330$ remaining pages to read $(R={330})$. Let's plug these values into the equation to find the value of $B$.

$\qquad B=55\cdot{4}+{330}=550$

Therefore, the book is $550$ pages long.

In [15]:
solveset(eq.subs([(B, 550), (R, 0)]))

{10}

To find how long it took Naoya to finish the entire book, we can plug $R=0$ into the equation and solve for $T$.

$\qquad\begin{aligned}550&=55T+0\\ 55T&=550\\ T&=10\end{aligned}$

The book is $550$ pages long.
It took Naoya $10$ hours to read the entire book.

## Charging the battery

A battery was charged. When the charging began, it was $23$ percent full. After $30$ minutes of charging, the battery was $89$ percent full.

**What percent was the battery charged per minute?**

**How many minutes did it take the battery to be fully charged?**

Let's say that the percentage of the battery's capacity that is charged increases by $V$ percent each minute. Then, it increases by $V\cdot T$ percent in $T$ minutes. In addition, we know that when the charging began, the battery was $23$ percent full.

The percentage of the battery's capacity that was charged at any given time is found by taking the percentage the battery was already charged when the charging began and adding to it the percentage that was additionally charged since then. We can express this with the equation $C=23+V\cdot T$, where:

* $C$ represents the percentage of the battery's capacity that is charged at a given time
* $V$ represents the charging rate (in percent per minute)
* $T$ represents the time (in minutes)

In [18]:
C, V, T = symbols('C V T')
eq = Eq(C, 23 + V * T)

We know that after $30$ minutes $(T={30})$, the battery was $89$ percent full $(C={89})$. Let's plug these values into the equation to find the value of $V$.

In [21]:
solveset(eq.subs([(C, 89), (T, 30.0)]))

{2.2}

$\qquad \begin{aligned}{89}&=23+V\cdot{30}\\ 30V&=66\\ V&=2.2\end{aligned}$

Therefore, the battery was charged at a rate of $2.2$ percent per minute.

To find how long it took the battery to be fully charged, we can plug $C=100$ into the equation and solve for $T$.

In [25]:
solveset(eq.subs([(C, 100), (V, 2.2)]))

{35.0}

$\qquad\begin{aligned}100&=23+2.2T\\ 2.2T&=77\\ T&=35\end{aligned}$

* **The battery was charged at a rate of $2.2$ percent per minute.**

* **It took the battery $35$ minutes to be fully charged.**

## Vlad and his mighty oak

When Vlad moved to his new home a few years ago, there was a young oak tree in his backyard. He measured it once a year and found that it grew by $26$ centimeters each year. $4.5$ years after he moved into the house, the tree was $292$ centimeters tall.

**How many centimeters tall was the tree when Vlad moved into the house?**
 
**How many years passed from the time Vlad moved in until the tree was $357$ centimeters tall?**

The tree grew by $26$ centimeters each year, so it grew by $26T$ centimeters in $T$ years.

The tree's height at a given time is found by taking its height when Vlad moved to his new home and adding to it the length it grew since then. We can express this with the equation $H=A+26T$, where:

* $H$ represents the tree's height at a given time (in centimeters)
* $A$ represents the tree's original height (in centimeters)
* $T$ represents the time (in years)

In [27]:
H, A, T = symbols('H A T')
eq = Eq(H, A + 26 * T)

We want to find $A$, so let's first solve the equation for $A$:

In [28]:
solveset(eq, A)

{H - 26*T}

$\qquad\begin{aligned}H&=A+26T\\ A&=H-26T\end{aligned}$

Now, we know that after $4.5$ years $(T={4.5})$, the tree was $292$ centimeters tall $(H={292})$. Let's plug these values into the equation to find the value of $A$.

In [29]:
solveset(eq, A).subs([(H, 292), (T, 4.5)])

{175.0}

$\qquad A={292}-26\cdot{4.5}=175$

Therefore, when Vlad moved into the house, the tree was $175$ centimeters tall.

To find how long it took the tree to reach a height of $357$ centimeters, we can plug $H=357$ into the equation and solve for $T$.

In [32]:
solveset(eq.subs([(H, 357), (A, 175)]), T)

{7}

$\qquad\begin{aligned}175&=357-26T\\ 26T&=182\\ T&=7\end{aligned}$

* **When Vlad moved into the house, the tree was $175$ centimeters tall**

* **The tree reached a height of $357$ centimeters $7$ years after Vlad moved in.**

## Sean's slushie

Sean tried to drink a slushy as fast as he could. He drank the slushy at a constant rate. There were originally $275$ milliliters of slushy in the cup. After $13$ seconds, $210$ milliliters of slushy remained.

**How many milliliters per second did Sean drink?**

**How many seconds did it take Sean to drink all the slushy?**

In [4]:
y2, y1, x2, x1 = symbols('y2 y1 x2 x1')
x1 = 0
y1 = 275
x2 = 13
y2 = 210
m = (y2 - y1)/(x2 - x1)
m

-5.0

In [5]:
275/5

55.0

## Agent Hunt's files

Agent Hunt transferred classified files from the CIA mainframe onto his flash drive. The drive had some files on it before the transfer, and the transfer happened at a rate of $4.4$ megabytes per second. After $32$ seconds, there were $384$ megabytes on the drive. The drive had a maximum capacity of $1000$ megabytes.

**How many megabytes full was the drive when the transfer began?**

The files were transferred at a rate of $4.4$ megabytes per second, so $4.4T$ megabytes were transferred in $T$ seconds.

The size of the files on the drive is comprised of the files that were on the drive before the transfer began and the files that were transferred. We can express this with the equation $S=A+4.4T$, where:

* $S$ represents the size of the files on the drive at a given time (in megabytes)

* $A$ represents the size of the files that were on the drive before the transfer began (in megabytes)

* $T$ represents the time (in seconds)

In [7]:
S, A, T = symbols('S A T')
eq = Eq(S, A + 4.4 * T)

We want to find $A$, so let's first solve the equation for $A$:

In [8]:
solveset(eq, A)

{S - 4.4*T}

$\qquad\begin{aligned}S&=A+4.4T\\ A&=S-4.4T\end{aligned}$

Now, we know that after $32$ seconds $(T={32})$, there were $384$ megabytes on the drive $(S={384})$. Let's plug these values into the equation to find the value of $A$.

In [10]:
eq = Eq(A, S - 4.4 * T)
eq.subs([(T, 32), (S, 384)])

Eq(A, 243.2)

$\qquad A={384}-4.4\cdot{32}=243.2$

Therefore, when the transfer began, the drive had $243.2$ megabytes on it.

**How many seconds from the time that Agent Hunt started the transfer did it take the drive to be completely full?**

Plug $S=1000$ into the equation and solve for $T$:

In [14]:
solveset(eq.subs([(A, 243.2), (S, 1000)]))

{172.0}

$\qquad\begin{aligned}243.2&=1000-4.4T\\ 4.4T&=756.8\\ T&=172\end{aligned}$

**It took the drive $172$ seconds to be completely full.**